src/io/utilities/readSBML.m

```diff
@@ -636,10 +640,10 @@
     model.lb = fbc_lb;
     model.ub = fbc_ub;
     if noObjective==0; % when there is an objective function
-        indexObj=findRxnIDs(model,fbc_obj);
+        indexObj=findRxnIDs(model,cleanUpFormatting(fbc_obj));
         % indexObj=find(strcmp(fbc_obj,model.rxns))
-        model.c(indexObj)=1;
-        model.osense = - sign(fbc_obj_value);
+        model.c(indexObj) = fbc_obj_value(:);
+        model.osense = fbc_obj_sense;
```

Comment:
Is osense supposed to be a single number '-1' for 'max' and '1' for 'min'? That is what I assumed from the solver functions, e.g. solveCobraLP.
This is implemented in the suggested update by defining the addition fbc_obj_sense.

In the current master version, 
```
fbc_obj_value = modelSBML.(fbc_list{f})(1).fbc_type(1:3);
...
model.osense = - sign(fbc_obj_value);
```
The fbc_obj_value is either 'min' or 'max'. "- sign(fbc_obj_value)" is always [-1 -1 -1] in either case. (I tried loading some models with fbc_version using R2015b. Or am I missing anything here?).

If using `model.osense = - sign(fbc_obj_value);`
with the updated definition of fbc_obj_value from line 306, osense will become a vector with 1 or -1 corresponding to each non-zero element in the objective vector.